terra/executor/resources.py

```diff
@@ -334,8 +404,16 @@ def register_resource(cls, name, *args, **kwargs):
     ValueError
         If a resource with that name already exists
     '''
+    '''
```

Comment:
why end and then re-start the `'''`

Reply:
I'm commenting this in case we ever need to add this feature. Right now we are expecting the overhead of acquiring a resource to be a fraction of a second. But as soon as we find a scenario when that's true, I want this code to be ready to pop in, and this comment is part of that.